### PR TITLE
Refactor telegram channel

### PR DESCRIPTION
**Pull request by filintod**
**Proposed changes**:
- I noticed there were two log messages enter after a return in TelegramInput.set_webhook that would never get executed. I also change logger.warn with logger.warning as the warn method is set for deprecation for a while.

After looking at the code I also saw some repeated code in TelegramOutput.send_text_with_buttons that could be simplified

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

**Comment by amn41** [comment]
thanks for creating this PR! @akelad will review it. Please do also sign the contributor's agreement, otherwise we won't be able to merge the PR.

File: rasa_core/channels/telegram.py [hunk 1]
```diff
@@ -40,20 +40,10 @@ def send_text_with_buttons(self, recipient_id, text,
...
         :button_type custom: custom keyboard
         """
-        if button_type == "inline":
+        if button_type in ("inline", "vertical"):
             button_list = [[InlineKeyboardButton(s["title"],
                             callback_data=s["payload"]) for s in buttons]]
             reply_markup = InlineKeyboardMarkup(button_list)
-            return self.send_message(recipient_id, text,
-                                     reply_markup=reply_markup)
-        if button_type == "vertical":
```
[the remark this answers is not in the document]
Comment (filintod) [reply rasa_core/channels/telegram.py]: I did not get rid of it I just put it together with "inline" as the code was pretty much identical.

[the remark this answers is not in the document]
Comment (akelad) [reply rasa_core/channels/telegram.py]: No, there's an important difference that the "inline" button is a list of buttons, whereas the "vertical" button is a list of lists of buttons. That's the telegram format 😄 
so:
``[[InlineKeyboardButton(s["title"], callback_data=s["payload"]), InlineKeyboardButton(s["title"], callback_data=s["payload"])]]``
vs
``[[InlineKeyboardButton(s["title"],  callback_data=s["payload"])], [InlineKeyboardButton(s["title"],  callback_data=s["payload"])]]``

[the remark this answers is not in the document]
Comment (filintod) [reply rasa_core/channels/telegram.py]: I see my mistake just fixed it now.

**Pull request merged**